editor/src/core/commenting/comment-maintainer.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { MultiplayerWrapper } from '../../utils/multiplayer-wrapper'
+import { getThreadLocationOnCanvas, useCanComment, useScenes } from './comment-hooks'
+import {
+  isSceneThreadMetadata,
+  useEditThreadMetadata,
+  useThreads,
+} from '../../../liveblocks.config'
+import { getIdOfScene } from '../../components/canvas/controls/comment-mode/comment-mode-hooks'
+import * as EP from '../shared/element-path'
+import { isNotNullFiniteRectangle } from '../shared/math-utils'
+
+export const CommentMaintainer = React.memo(() => {
+  const canComment = useCanComment()
+
+  if (!canComment) {
+    return null
+  }
+
+  return (
+    <MultiplayerWrapper errorFallback={null} suspenseFallback={null}>
+      <CommentMaintainerInner />
+    </MultiplayerWrapper>
+  )
+})
+CommentMaintainer.displayName = 'CommentMaintainer'
+
+const CommentMaintainerInner = React.memo(() => {
+  useMaintainComments()
+
+  return null
+})
+CommentMaintainerInner.displayName = 'CommentMaintainerInner'
+
+function useMaintainComments() {
+  const { threads } = useThreads()
+  const scenes = useScenes()
+  const editThreadMetadata = useEditThreadMetadata()
+
+  threads.forEach(async (t): Promise<void> => {
```

Comment:
I was about to ask if subsequent calls to this can trample over each other, but I don't _think_ they will because the promises would all be queued up in order.